src/app/app.component.ts

```diff
@@ -41,57 +38,64 @@ export class AppComponent implements OnInit {
   coordinates: { latitude: number; longitude: number }[] = [];
   finalDestinationName = '';
   weatherCoordinates!: WeatherCoordinates;
-
+  cityName = 'Herrljunga';
+  stopPressed = false;
   private handleStopListCounter = 0;
   private previousStopList: any[] = [];
-  stopPressed = false;
+  weatherDataArray: any[] = [];
+
 
   constructor(
     private apiService: ApiService,
-    private coordinatesService: CoordinatesService,
     private stopListService: StopListService,
     private stopButtonService: StopButtonService,
+    private locationService: LocationService,
   ) {}
 
   //stop buton
   handleButtonStop() {
     this.stopButtonService.notifyButtonClick();
   }
-
   handleButtonStopClear() {
     this.stopButtonService.notifyClearButtonClick();
   }
 
   ngOnInit(): void {
     this.initConnection();
-    this.getWeatherCoordinates(this.latitude, this.longitude);
+    this.fetchWeatherByCityName(this.cityName);
   }
-
-  // get weather coordinats
-  getWeatherCoordinates(latitude: number, longitude: number): void {
-    this.apiService
-      .getWeatherCoordinates(latitude, longitude)
-      .subscribe((data) => {
-        this.weatherCoordinates = data;
-        this.weatherTemperature = this.weatherCoordinates.temp;
-        this.weatherWind = this.weatherCoordinates.winSpd;
-        if (this.weatherCoordinates && this.weatherCoordinates.wsymb) {
-          this.weatherIconValue = this.weatherCoordinates.wsymb;
+  fetchWeatherByCityName(cityName: string): void {
+    this.apiService.getWeatherCityName(cityName).subscribe({
+      next: (data) => {
+        if (data) {
+          this.weatherDataArray.push(data);
+          console.log('API weather:', data);
         }
-        console.log('Weather', data);
-      });
+      },
+      error(err) {
+        console.error('Something wrong occurred: ' + err);
+      },
+      complete() {
```

Comment:
When doing a regular HTTP call we always only get one response. So the first response is always also the last respone (there is only one!). So no need to distinguish between `next()` and `done()` on HTTP calls.

If we have a Subject that sends multiple events (like keyboard presses) then we must use `next()` and `complete()` if we want to handle some closing logic.